### ISSUE TITLE
About human brain data and pre-trained model

### ISSUE THREAD

**Issue by gaunny** [AxelElaldi/e3so3_conv#1]
Hi!
I am very interested in this work and I try to train/test on other dMRI data.  I tried to train on FiberCup data, ISMRM2015 data and my private human brain data respectively. However, the value of loss function is always high,  especially for FiberCup data and my private human brain data, and the fODFs of ESD and RT-ESD obtained are very poor.
In this command **python train.py --data_path $your_data_path --batch_size 32 --lr 0.0017 --epoch 50 --filter_start 1 --sh_degree 18 --save_every 1 --loss_intensity L2 --loss_sparsity cauchy --loss_non_negativity L2 --sigma_sparsity 1e-05 --sparsity_weight 1e-06 --intensity_weight 1 --nn_fodf_weight 1 --pve_weight 1e-11 --wm --gm  --rf_name $your_rf_algo_choice --depth 5 --patch_size 5 --normalize --kernel_sizeSpa 3 --conv_name mixed** , I wonder if the parameters set in this command apply to all data? It does not appear to be suitable for FiberCup and other data. If possible, can you provide your specific training data and pre-trained model? 
Thank you very much and look forward to your reply.

**Comment by AxelElaldi** [comment]
Hello!
These parameters should work correctly in most cases, assuming the training/testing data is B0-normalized. Let me know if this pre-processing step improves your results!

**Comment by gaunny** [comment]
Yes, I make the data B0-normalized and it works better. But I find that the value of the Loss hardly changes, as shown in the figure, and often the early stopping criterion is reached after training for only 8-9 epochs (both for Fibercup and ismrm2015 data). The estimated fODF is also not completely accurate compared to CSD.

![图片1](https://github.com/AxelElaldi/e3so3_conv/assets/48590551/93479fcc-0df8-4bfc-9a9c-7ec30f49b22e)
![图片2](https://github.com/AxelElaldi/e3so3_conv/assets/48590551/297e2137-ef51-4a09-8753-b5bcfd6b05d6)

**Comment by AxelElaldi** [comment]
Thanks for sharing your results! I forgot to write that the tissue response functions should be computed after the B0 normalization. 

You are correct, the convergence can happen early during the training, and decrease only slightly after the first or second epoch. In my ISMRM2015 experiment, the intensity loss decreases to something in the range of ~ (3 - 5) x 1e-3 after the first epoch. Here is an old plot from Tensorboard:
![image](https://github.com/AxelElaldi/e3so3_conv/assets/45570710/85567168-4852-4f2a-86e4-78156256fb18)

**FiberCup:**
Unfortunately, I didn't try the FiberCup data myself. From your shared loss history, the intensity loss is higher than expected.
- Do you use the three provided shells (650/1500/2000) or only a single shell?
- The image dimension is 64x64x3 while the --patch_size option is set to 5. RT-ESD has a smoothing effect on the fODF of a patch, and thus the low size of the data on the z-axis may adversely affect the results as it will incorporate 0-valued signals. Could you try lowering the --patch_size to 3 and look at the middle slice?
- From the images you shared, it seems the bvecs are flipped on the x-axis. It should not impact the result, but flipping the x-coordinate of the bvecs table before running csd and rtesd can help with the visualization. If you use MITK and want to keep the bvecs table as it is, you can flip the fODF x-axis on the visualization panel.

**Comment by gaunny** [comment]
Thanks for your help! In FiberCup, I use only one shell (b=2000) to train and I also use the tissue response functions computed after the B0 normalization. When I change the patch size to 3 and 2 respectively, the results show better, and I find 3 is best.
![图片3](https://github.com/AxelElaldi/e3so3_conv/assets/48590551/ca9150cb-e6d3-4e0b-b0b5-93e6c475cb59)
This result is significantly better than CSD.
As for the image being inverted, I found that it is MITK problem, using MRtrix open does not have this problem. 
Thank you!
